react/week2/todo/src/Todo.js

```diff
@@ -0,0 +1,81 @@
+
+import React, { useState } from 'react';
+import Item from './Item';
+import TodoForm from "./TodoForm";
+
+const Todo = () => {
+
+
+	const taskItem = [
+		{
+			text: 'Get out of bed',
+			isCompleted: false
+		},
+
+		{
+			text: 'Brush teeth',
+			isCompleted: false
+		},
+
+		{
+			text: 'Eat breakfast',
+			isCompleted: false
+		},
+	];
+
+	/**
+	 * todos: is taskItem, whose value will be equal to taskItem that we pass in useState() as a parameter
+	 * setTodos is like setState
+	 */
+	const [ todos, setTodos ] = useState( taskItem );
+
+	const addToDo = ( text ) => {
+		const newToDos = [ ...todos, { text } ];
+		setTodos( newToDos );
+	};
+
+	const handleItemClick = ( index ) => {
+		// Get all todos array from state.
+		const newTodos = [ ...todos ];
+
+		// Set isCompleted property to reverse of what its current value is ( boolean )
+		newTodos[ index ].isCompleted = ! newTodos[ index ].isCompleted;
+
+		// Set State with the new array of todos with the updated value
+		setTodos( newTodos );
+	};
+
+	const handleRemoveClick = ( index ) => {
+		// Get all todos array from state.
+		const newTodos = [ ...todos ];
+
+		// Remove the clicked item from the todos array
+		newTodos.splice( index, 1 );
+
+		// Set State with the new array of todos with the updated value
+		setTodos( newTodos );
+	};
+
+
+	return (
+		<div className="todo-container">
+			<h1 className="main-heading">Todo App</h1>
+			<TodoForm addToDo={addToDo}/>
+			<div className="todo-list">
+				{ todos.length ? (
```

Comment:
I'm not sure if there is any need of checking the length and returning an empty string if the array is empty. You can drop the whole logic (leave only the map itself) and the code will seemingly work the same way.